### DISCUSSION TITLE
So about globals...

### DISCUSSION THREAD

**Discussion by aoqia194** [YimMenu/YimMenu#384] [Q&A] [answered]
Sorry I keep asking about stuff related to GTA here that I could probably find if I looked for another 4 hours but I've been trying to look for an hour or so on how globals are actually accessed. I know how to find them in decompiled game scripts. And is the globals menu in debug a working implementation of it already?
I am asking this mainly to gain insight on how the game works (as I know little currently) and also because I'm planning to add some useful features for myself in a fork (not going to pr because these are [basic features](https://github.com/faithfullunaa/YimMenu/tree/stat_editor))
Any response is helpful 🍉

**Comment by Yimura** [comment, marked as answer]
Globals are just a large array of bytes in memory.

If you take a look at pointers.cpp you can see that we get the pointer for where all the data is stored for of the scripts.
https://github.com/YimMenu/YimMenu/blob/9d48b02d632f40427314a48b49b132014d6a6e93/BigBaseV2/src/pointers.cpp#L69-L72

Then in script_global.cpp you can see how from an global its index we calculate where in the byte array the global starts.
https://github.com/YimMenu/YimMenu/blob/9d48b02d632f40427314a48b49b132014d6a6e93/BigBaseV2/src/script_global.cpp#L22-L25
I'm assuming these values to get the address of global is done similarly somewhere in the game its assembly.

Once you've got the starting address of your specific global you can do anything with it, cast it to an int*, const char* or even an entire class structure anything is possible.
